AssertionError: invalid argspec BoundMethodArgSpecWrapper

```py
Traceback (most recent call last):
  File "/v/site-packages/pyanalyze/name_check_visitor.py", line 705, in visit
    ret = method(node)
  File "/v/site-packages/pyanalyze/name_check_visitor.py", line 2843, in visit_Subscript
    return_value, _ = self._get_argspec_and_check_call(
  File "/v/site-packages/pyanalyze/name_check_visitor.py", line 3372, in _get_argspec_and_check_call
    extended_argspec = self._get_argspec_from_value(callee_wrapped, node)
  File "/v/site-packages/pyanalyze/name_check_visitor.py", line 3465, in _get_argspec_from_value
    return self._get_argspec(callee_wrapped.val, node, name=name)
  File "/v/site-packages/pyanalyze/name_check_visitor.py", line 3495, in _get_argspec
    return self.arg_spec_cache.get_argspec(obj, name=name, logger=self.log)
  File "/v/site-packages/pyanalyze/arg_spec.py", line 1124, in get_argspec
    argspec = self._cached_get_argspec(obj, kwargs)
  File "/v/site-packages/pyanalyze/arg_spec.py", line 1136, in _cached_get_argspec
    extended = self._uncached_get_argspec(obj, kwargs)
  File "v/site-packages/pyanalyze/arg_spec.py", line 1161, in _uncached_get_argspec
    return BoundMethodArgSpecWrapper(argspec, KnownValue(obj.__self__))
  File "/v/site-packages/pyanalyze/arg_spec.py", line 142, in __init__
    assert isinstance(argspec, ExtendedArgSpec), "invalid argspec %r" % (argspec,)
AssertionError: invalid argspec BoundMethodArgSpecWrapper(
  argspec=ExtendedArgSpec(
    _has_return_value=False,
    arguments=[Parameter(default_value=<object object at 0x7fa6a833f990>, name='self', typ=None)],
    implementation=None,
    kwargs='kwargs',
    kwonly_args=None,
    name='GenericAlias',
    params_of_names={
      'self': Parameter(default_value=<object object at 0x7fa6a833f990>, name='self', typ=None),
      'args': Parameter(default_value=<object object at 0x7fa6a833f990>, name='args', typ=TypedValue(typ=<class 'tuple'>)),
      'kwargs': Parameter(default_value=<object object at 0x7fa6a833f990>, name='kwargs', typ=TypedValue(typ=<class 'dict'>))},
    return_value=UnresolvedValue(), starargs='args'),
    self_value=TypedValue(typ=<class 'types.GenericAlias'>))

Internal error: AssertionError(-"-) (code: internal_error)
```

I'm sorry the code is closed-source, I could try to track it down somehow...
I get this error using `Python 3.9.0b1` but not 3.8

Full MRE:

```py
from queue import Queue
from typing import List

class I:
    ...

class X:
    def __init__(self):
        self.q: Queue[List[I]] = Queue()
```

mypy tells me that I have `error: syntax error in type comment`, perhaps I can't type 😮 

pyanalyze gives off a traceback `AssertionError: invalid argspec BoundMethodArgSpecWrapper(argspec=ExtendedArgSpec(_has_return_value=False, arguments=[Parameter(default_value=<object object at 0x7f8c801cf990>, name='self', typ=None)], implementation=None, kwargs='kwargs', kwonly_args=None, name='GenericAlias', params_of_names={'self': Parameter(default_value=<object object at 0x7f8c801cf990>, name='self', typ=None), 'args': Parameter(default_value=<object object at 0x7f8c801cf990>, name='args', typ=TypedValue(typ=<class 'tuple'>)), 'kwargs': Parameter(default_value=<object object at 0x7f8c801cf990>, name='kwargs', typ=TypedValue(typ=<class 'dict'>))}, return_value=UnresolvedValue(), starargs='args'), self_value=TypedValue(typ=<class 'types.GenericAlias'>))`

--- a/pyanalyze/annotations.py
+++ b/pyanalyze/annotations.py
@@ -32,11 +32,16 @@ from .value import (
 )
 
 try:
-    from typing import get_origin  # Python 3.9
+    from typing import get_origin, get_args  # Python 3.9
+    from types import GenericAlias
 except ImportError:
+    GenericAlias = None
 
     def get_origin(obj):
         return None
+
+    def get_args(obj):
+        return ()
 
 
 def type_from_ast(ast_node, visitor=None):
@@ -110,6 +115,10 @@ def _type_from_runtime(val, ctx):
         origin = typing_inspect.get_origin(val)
         args = typing_inspect.get_args(val)
         return _value_of_origin_args(origin, args, val, ctx)
+    elif GenericAlias is not None and isinstance(val, GenericAlias):
+        origin = get_origin(val)
+        args = get_args(val)
+        return GenericValue(origin, [_type_from_runtime(arg, ctx) for arg in args])
     elif isinstance(val, type):
         if val is type(None):
             return KnownValue(None)

--- a/pyanalyze/arg_spec.py
+++ b/pyanalyze/arg_spec.py
@@ -139,6 +139,8 @@ class BoundMethodArgSpecWrapper(InspectableClass):
     """
 
     def __init__(self, argspec, self_value):
+        if isinstance(argspec, BoundMethodArgSpecWrapper):
+            argspec = argspec.argspec
         assert isinstance(argspec, ExtendedArgSpec), "invalid argspec %r" % (argspec,)
         self.argspec = argspec
         self.self_value = self_value
@@ -925,8 +927,8 @@ class ArgSpecCache(object):
             [
                 Parameter("expected"),
                 Parameter("actual"),
-                Parameter("message", default_value=None,),
-                Parameter("extra", default_value=None,),
+                Parameter("message", default_value=None),
+                Parameter("extra", default_value=None),
             ],
             name="assert_is",
             implementation=_assert_is_impl,
@@ -935,8 +937,8 @@ class ArgSpecCache(object):
             [
                 Parameter("expected"),
                 Parameter("actual"),
-                Parameter("message", default_value=None,),
-                Parameter("extra", default_value=None,),
+                Parameter("message", default_value=None),
+                Parameter("extra", default_value=None),
             ],
             name="assert_is_not",
             implementation=_assert_is_not_impl,

--- a/pyanalyze/name_check_visitor.py
+++ b/pyanalyze/name_check_visitor.py
@@ -2105,14 +2105,14 @@ class NameCheckVisitor(node_visitor.ReplacingNodeVisitor):
         allow_call = method not in self.config.DISALLOW_CALLS_TO_DUNDERS
         with self.catch_errors() as left_errors:
             left_result = self._check_dunder_call(
-                source_node, left, method, [right], allow_call=allow_call,
+                source_node, left, method, [right], allow_call=allow_call
             )
         if not left_errors:
             return left_result
 
         with self.catch_errors() as right_errors:
             right_result = self._check_dunder_call(
-                source_node, right, rmethod, [left], allow_call=allow_call,
+                source_node, right, rmethod, [left], allow_call=allow_call
             )
         if not right_errors:
             return right_result
@@ -2871,7 +2871,7 @@ class NameCheckVisitor(node_visitor.ReplacingNodeVisitor):
         lookup_val = callee_val.get_type_value()
 
         def on_error(typ):
-            self._show_error_if_checking(
+            self.show_error(
                 node,
                 "Object of type %s does not support %r" % (callee_val, method_name),
                 error_code=ErrorCode.unsupported_operation,

--- a/pyanalyze/test_annotations.py
+++ b/pyanalyze/test_annotations.py
@@ -436,7 +436,6 @@ def capybara(x: Type[str], y: "Type[int]"):
         self.assert_passes(
             """
 from queue import Queue
-from typing import List
 
 class I:
     ...

--- a/pyanalyze/test_annotations.py
+++ b/pyanalyze/test_annotations.py
@@ -431,7 +431,7 @@ def capybara(x: Type[str], y: "Type[int]"):
 """
         )
 
-    @skip_before((3, 6))
+    @skip_before((3, 9))
     def test_generic_alias(self):
         self.assert_passes(
             """

--- a/pyanalyze/test_annotations.py
+++ b/pyanalyze/test_annotations.py
@@ -222,7 +222,7 @@ class X:
     @abstractmethod
     def f(self) -> int:
         pass
-""",
+"""
         )
         self.assert_fails(
             ErrorCode.incompatible_return_value,
@@ -428,5 +428,24 @@ from typing import Type
 def capybara(x: Type[str], y: "Type[int]"):
     assert_is_value(x, SubclassValue(str))
     assert_is_value(y, SubclassValue(int))
+"""
+        )
+
+    @skip_before((3, 6))
+    def test_generic_alias(self):
+        self.assert_passes(
+            """
+from queue import Queue
+from typing import List
+
+class I:
+    ...
+
+class X:
+    def __init__(self):
+        self.q: Queue[I] = Queue()
+
+def f(x: Queue[I]) -> None:
+    assert_is_value(x, GenericValue(Queue, [TypedValue(I)]))
 """
         )

